packages/smooth_app/lib/pages/scan/continuous_scan_page.dart

```diff
@@ -124,60 +124,15 @@ class _ContinuousScanPageState extends State<ContinuousScanPage> {
     );
   }
 
-  Widget _buildButtonsRow(BuildContext context, ContinuousScanModel model) {
-    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
-    final ButtonStyle buttonStyle = ButtonStyle(
-      shape: MaterialStateProperty.all<RoundedRectangleBorder>(
-        RoundedRectangleBorder(
-          borderRadius: BorderRadius.circular(18.0),
-        ),
-      ),
-    );
-    return AnimatedOpacity(
-      opacity: areButtonsRendered(model) ? 0.8 : 0.0,
-      duration: const Duration(milliseconds: 50),
-      child: Padding(
-        padding: const EdgeInsets.symmetric(
-            vertical: VERY_SMALL_SPACE, horizontal: MEDIUM_SPACE),
-        child: Row(
-          mainAxisAlignment: MainAxisAlignment.spaceBetween,
-          children: <Widget>[
-            ElevatedButton.icon(
-              style: buttonStyle,
-              icon: const Icon(Icons.cancel_outlined),
-              onPressed: model.clearScanSession,
-              label: Text(appLocalizations.clear),
-            ),
-            ElevatedButton.icon(
-              style: buttonStyle,
-              icon: const Icon(RankingFloatingActionButton.rankingIconData),
-              onPressed: () => _openPersonalizedRankingPage(context),
-              label: Text(
-                appLocalizations.plural_compare_x_products(
-                  model.getBarcodes().length,
-                ),
-              ),
-            ),
-          ],
-        ),
-      ),
-    );
+  void setupScanner(QRViewController controller) {
+    _controller = controller;
+    _controller?.scannedDataStream
+        .listen((Barcode barcode) => _model.onScan(barcode.code));
```

Comment:
I think you should keep the whole `Barcode`, not just the code. This way we can prevent @teolemon's bug when he scanned a QR code.
Or, you should not call `_model.onScan` when the barcode is not a barcode.

Reply:
I had it at the beginning so that I passed the barcode object on, but this was very confusing in the model, because both libraries have a barcode class, which means I always had to work with `google_ml_kit_barcode_scanner.Barcode` and `qr_code_scanner.Barcode`. 
This PR is already big enouph let's just introduce a non-barcode check in a different one.